src/mesh/difops.cxx

```diff
@@ -287,23 +297,21 @@ const Field3D Grad_par_CtoL(const Field3D &var) {
 
   if (var.hasYupYdown()) {
     // NOTE: Need to calculate one more point than centred vars
-    for(int jx=0; jx<mesh->LocalNx;jx++) {
-      for(int jy=1;jy<mesh->LocalNy;jy++) {
-        for(int jz=0;jz<mesh->LocalNz;jz++) {
-          result(jx, jy, jz) = 2.*(var(jx, jy, jz) - var.ydown()(jx, jy-1, jz)) / (metric->dy(jx, jy) * sqrt(metric->g_22(jx, jy)) + metric->dy(jx, jy-1) * sqrt(metric->g_22(jx, jy-1)));
-      }
-      }
+    BOUT_FOR(i, var.getRegion("RGN_NOY")) {
```

Comment:
I'm not sure the RGN_NOY is really what we want here -- the original loop does all of y except for `jy=0`.

Secondly do we actually need `Grad_par_CtoL`? The majority of this looks like it should be possible to use `DDY` with an appropriate input and output location.

Reply:
@bendudson would know for sure, but I think `Grad_par_CtoL`, etc., were an earlier implementation of staggered grids, before `Field3D`s had `location`s.

[Looks like `Grad_par_CtoL` did have better handling of the `dy`/metric terms than we used to in `DDY` before the recent updates to have staggered `Coordinates`. I'd not realised that before! But `DDY` should have better handling now because `dy` and `g_22` are interpolated more accurately (4-point stencil instead of 2-point).]

I'd be in favour of deprecating `Grad_par_CtoL`, etc., now so we don't have to keep maintaining them.

Reply:
Thanks, I'd certainly be happy to deprecate as much special case code as possible if we can rewrite/replace using more standard/core code.